### ISSUE TITLE
CheckboxSetMultivalueField can't save values

### ISSUE THREAD

**Issue by michalkleiner** [silverstripe-australia/silverstripe-multivaluefield#73]
Not sure when this started happening, couldn't find an exact change related to this, could be 3.1 to 5.0 change, but when trying to save a set of values the abstract underlying `MutliSelectField` complains that it expects a string for the `stringDecode` method when it's provided an array.

## PRs
- #87

**Comment by xini** [comment]
@michalkleiner have you ever found a solution for this?

**Comment by kevingroeger** [comment]
I stumbled upon this issue recently, when I tried to figure out why it doesn't work for me.

In case anyone else still has this issue as well, here is how you can fix it temporarily, until the PR is merged:

```
<?php

namespace Extension;

use SilverStripe\ORM\DataObjectInterface;
use Symbiote\MultiValueField\Fields\MultiValueCheckboxField;

/**
 * Fixes an issue with the MultiValueCheckboxField,
 * where it errors out because it can't parse its own values
 *
 * Temporary fix until the following PR gets merged:
 * https://github.com/symbiote/silverstripe-multivaluefield/pull/87
 */
class MultiCheckBoxField extends MultiValueCheckboxField
{
    /**
     * Load the value from the DataObject into this field
     *
     * @param DataObjectInterface $record
     */
    public function loadFrom(DataObjectInterface $record)
    {
        $fieldName = $this->getName();
        if (empty($fieldName) || empty($record)) {
            return;
        }

        if ($record->hasField($fieldName)) {
            $value = $record->$fieldName;

            parent::setValue($value);
        }
    }
}
```

Instead of using `MultiValueCheckboxField::create(...)` in your Code, you then can use `MultiCheckBoxField::create(...)` and it should work!

**Comment by michalkleiner** [comment]
Thanks for the PR @xini and thanks for confirming the code in the PR works for you, @kevingroeger. It would be great if we could have this covered by a test so that we know if/when something breaks it.

**Comment by xini** [comment]
@michalkleiner I'm working on Windows and for some reason the config isn't loaded from class statics when I run unit tests. Like that I can't really write any test I'm afraid. Any idea why that might be? I had the same issue here; https://github.com/silverstripe/silverstripe-framework/pull/10688#pullrequestreview-1319048343

**Comment by michalkleiner** [comment]
Sorry, I'm on a unix system and use docker for web stuff anyway, so can't help with Windows. But we could create a docs repo issue to improve the documentation/tutorials/instructions on running/contributing tests on different operating systems.

**Comment by xini** [comment]
> But we could create a docs repo issue to improve the documentation/tutorials/instructions on running/contributing tests on different operating systems.

Thanks. That's if we find out what the issue is, yes.

**Comment by xini** [comment]
@michalkleiner I have now added tests to #87.

**Comment by kevingroeger** [comment]
The changes from @xini seem to be working just fine :) Would highly appreciate if it could get merged so I can get rid of my hacky solution :D 

@michalkleiner

**Comment by michalkleiner** [comment]
Thanks @xini and @kevingroeger for your time and contributions, this has now been merged and released in [5.4.1](https://github.com/symbiote/silverstripe-multivaluefield/releases/tag/5.4.1) and [6.0.1](https://github.com/symbiote/silverstripe-multivaluefield/releases/tag/6.0.1)
